### PR TITLE
Convert setup_sunsim, sunsim, and dasslsim to using keyword arguments for reltol and abstol

### PR DESCRIPTION
**Pull request by iraikov**
This pull request converts setup_sunsim, sunsim, and dasslsim to using keyword arguments, updated all examples that call setup_sunsim and adds two more neural model examples.

### CODE REVIEW

File: examples/neural/exp2syn.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+function Exp2Syn(v,s)
+    @equations begin
+        der(s) = alpha * k_(v) * (1-s) - beta * s
+    end
+end    
+
+
+(component (type post-synaptic-conductance) (name AMPA)
```
Comment (tshort) [inline examples/neural/exp2syn.jl]: Did you mean to leave this stuff in here?

File: src/sundials.jl [hunk 2]
```diff
@@ -194,7 +194,7 @@ function sunsim(smem::SimSundials, tstop::Float64, Nsteps::Int)
     end
     SimResult(yout, [sm.outputs[yidx]])
 end
-sunsim(ss::SimState, tstop::Float64, Nsteps::Int) = sunsim(setup_sunsim(ss,1e-4,1e-4), tstop, Nsteps)
+sunsim(ss::SimState, tstop::Float64, Nsteps::Int) = sunsim(setup_sunsim(ss,reltol=1e-4,abstol=1e-4), tstop, Nsteps)
```
Comment (tshort) [inline src/sundials.jl]: It'd be nice to also have a version of `sunsim` with reltol and abstol keyword arguments, too.

**Comment by tshort** [comment]
Good to merge when you're ready.

**Comment by iraikov** [comment]
I think I managed to add two more commits that add keywords args to sunsim and dasslsim and remove the stray code, but please double check.

**Comment by tshort** [comment]
Looks good. Go ahead and merge it.

**Pull request merged**